pkg/manifests/manifests.go

```diff
@@ -1333,6 +1346,32 @@ func (f *Factory) PrometheusK8s(host string, grpcTLS *v1.Secret, trustedCABundle
 		},
 	})
 
+	if f.config.ClusterMonitoringConfiguration.GrafanaConfig.IsEnabled() {
+		// Add the secret volume.
+		p.Spec.Volumes = append(p.Spec.Volumes, v1.Volume{
+			Name: "secret-prometheus-k8s-htpasswd",
+			VolumeSource: v1.VolumeSource{
+				Secret: &v1.SecretVolumeSource{
+					SecretName: "prometheus-k8s-htpasswd",
+				},
+			},
+		})
+
+		// Add the htpasswd arg and volume mount to the proxy container.
+		for i := range p.Spec.Containers {
```

Comment:
it might be clearer to move it to the range loop above where we alreayd tweak the prometheus-proxy container?

Reply:
Done.